### PR TITLE
feat(web): builtin infobox video and markdown block

### PR DESCRIPTION
**Pull request by airslice**
# Overview

This PR adds two new builtin infobox block: VideoBlock and MarkdownBlock

## Memo

image

## Summary by CodeRabbit

- **New Features**
	- Introduced two new infobox block types: Video and Markdown, allowing users to embed videos and Markdown content.
	- Enhanced infobox functionality by adding options for video playback and dynamic Markdown rendering.

- **Bug Fixes**
	- Improved accuracy in determining selected features based on layer data.

- **Style**
	- Updated image display settings for better visibility within containers.

- **Chores**
	- Upgraded the `react-player` library for potential performance and feature improvements.

### CODE REVIEW

File: server/pkg/builtin/manifest_ja.yml [hunk 1]
```diff
@@ -1347,6 +1347,34 @@ extensions:
...
+            title: 余白
+  markdownInfoboxBetaBlock:
+    name: MD Text
+    description: Infobox MD Text Block
+    propertySchema:      
+      default:
+        title: MD Text Block
+        fields:
+          src:
+            title: Content
```
Comment (airslice) [inline server/pkg/builtin/manifest_ja.yml:1377]: Could anybody help on the ja translation?

**Review by pyshx — APPROVED** [review]

File: web/src/beta/features/Visualizer/Crust/Infobox/Block/builtin/Markdown/index.tsx [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+      icon={block?.extensionId}
+      isSelected={isSelected}
+      propertyId={block?.propertyId}
+      property={block?.property}
+      {...props}>
+      {evaluatedSrc !== undefined ? (
+        <ReactMarkdown remarkPlugins={plugins} linkTarget="_blank">
+          {evaluatedSrc || ""}
+        </ReactMarkdown>
+      ) : null}
```
[the remark this answers is not in the document]
Comment (airslice) [reply web/src/beta/features/Visualizer/Crust/Infobox/Block/builtin/Markdown/index.tsx:34]: Already checked.

File: web/src/beta/features/Visualizer/Crust/Infobox/Block/builtin/Markdown/index.tsx [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+import { InfoboxBlock } from "../../../types";
+import useExpressionEval from "../useExpressionEval";
+
+const plugins = [gfm];
+
+const MarkdownBlock: FC<CommonBlockProps<InfoboxBlock>> = ({ block, isSelected, ...props }) => {
+  const src = useMemo(
+    () => block?.property?.default?.src?.value as ValueTypes["string"],
+    [block?.property?.default?.src],
+  );
```
[the remark this answers is not in the document]
Comment (airslice) [reply web/src/beta/features/Visualizer/Crust/Infobox/Block/builtin/Markdown/index.tsx:18]: It's okey to be undefined.

File: web/src/beta/features/Visualizer/Crust/Infobox/Block/builtin/useExpressionEval.ts [hunk 4]
```diff
@@ -24,7 +24,18 @@ export default (value: unknown | undefined) => {
...
+    // TODO: fix it on core
+    let selectedFeature = visualizer.current?.layers.selectedFeature();
+    const selectedLayer = visualizer.current?.layers.selectedLayer();
+
+    if (selectedLayer?.type === "simple" && selectedLayer?.data?.isSketchLayer) {
+      selectedFeature = selectedLayer.computed?.features.find(f => f.id === selectedFeature?.id);
+    } else {
+      selectedFeature = visualizer.current?.layers.selectedFeature();
+    }
+
```
[the remark this answers is not in the document]
Comment (airslice) [reply web/src/beta/features/Visualizer/Crust/Infobox/Block/builtin/useExpressionEval.ts]: Updated.

File: server/pkg/builtin/manifest_ja.yml [hunk 5]
```diff
@@ -1347,6 +1347,34 @@ extensions:
               custom: カスタム
           property_list:
             title: プロパティリスト
+      panel:
+        title: パネル設定
+        fields:
+          padding:
+            title: 余白
+  videoInfoboxBetaBlock:
```
[the remark this answers is not in the document]
Comment (airslice) [reply server/pkg/builtin/manifest_ja.yml:1355]: Can't find this issue.

**Review by mkumbobeaty — APPROVED** [review]

**Pull request merged**